test/test.js

```diff
@@ -1427,34 +1440,59 @@ test( "don't revalidate the field when pressing special characters", function()
 	errors( 0, "Normal keyup" );
 } );
 
-test( "#1508: Validation fails to trigger when next field is already filled out", function() {
+test( "#1508: Validation fails to trigger when next field is already filled out", function( assert ) {
 
 	// The next field is already filled out.
 	$( "#lastname" ).val( "something" );
+	$( "#box2" ).val( "something" );
 
 	var event     = $.Event( "keyup", { keyCode: 9 } ),
-		element   = $( "#firstname" ),
-		validator = $( "#testForm1" ).validate();
-
-	// Fill the first element
-	element.val( "abc" );
-
-	// Tab to the next field
-	element.blur();
-	$( "#lastname" ).trigger( event );
-	$( "#lastname" ).focus();
-
-	// Tab back to element
-	$( "#lastname" ).blur();
-	element.trigger( event );
-	element.focus();
+		validator = $( "#testForm1" ).validate(),
+		done      = assert.async(),
+		check     = function( element, nextE, value, remote ) {
+
+			// Fill the first element
+			element.val( "something" );
+
+			// Tab to the next field
+			element.blur();
+			nextE.trigger( event );
+			nextE.focus();
+
+			// Tab back to element
+			nextE.blur();
+			element.trigger( event );
+			element.focus();
+
+			// Change the content
+			element.val( value );
+			element.trigger( "keyup" );
+
+			// 'element' should be invalid
+			if ( remote ) {
+				setTimeout( function() {
```

Comment:
Please use asyncTest for now. We will upadte the entire codebase in a separate PR. Thx for the heads up.